examples/bri-3/src/bri/communication/agents/bpiMessages.agent.ts

```diff
@@ -0,0 +1,85 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { BpiMessage } from '../models/bpiMessage';
+import { BpiMessageType } from '../models/bpiMessageType.enum';
+
+import { NOT_FOUND_ERR_MESSAGE } from '../api/err.messages';
+import { BpiMessageStorageAgent } from './bpiMessagesStorage.agent';
+import { BpiSubject } from '../../identity/bpiSubjects/models/bpiSubject';
+import { BpiSubjectStorageAgent } from '../../identity/bpiSubjects/agents/bpiSubjectsStorage.agent';
+
+@Injectable()
+export class BpiMessageAgent {
+  constructor(
+    private bpiMessageStorageAgent: BpiMessageStorageAgent,
+    private bpiSubjectStorageAgent: BpiSubjectStorageAgent,
+  ) {}
+
+  public async getFromAndToSubjectsAndThrowIfNotExist(
```

Comment:
Can you elaborate what is happening in this function?
I'm of the opinion if we can extract this info in a single call, if possible

Reply:
Checking for existence of the from and to bpi subjects.